src/rules/no-commonjs.js

```diff
@@ -25,6 +25,11 @@ function allowRequire(node, options) {
   return options.allowRequire
 }
 
+function validateScope(scope) {
+  if (scope.variableScope.type === 'module') return true
```

Comment:
I think this might break the current behavior with try... catch. I'm AFK right now, will add a test case later.